### PR TITLE
use of gpio pins instead MCP io expander

### PR DESCRIPTION
**Pull request by moinmoin-sh**
I played with the `StateMonitor FW` and created a proposal that utilises an alternative of using different I/O hardware.
Here the MCPs are replaced with the 16 usable GPIOs of an ESP32 between GPIO2 and GPIO27 which are transformed into a 16-bit `uint16_t.` This mimics a configuration with one MCP. It offers the full functionality of the OXRS eco system, changes are in `main.cpp` only. None of the API has changed except the index which equals here the GPIO (-> 2,4,5, .... 26,27) for convenience. Shows again the flexibility of our architecture.
This allows to use a very simple piece of HW (just an ESP32 with the GPIOs broken out) to play with the OXRS eco system and get familiar with it. Could motivate more people to jump on. It could also be used to make a very simple `StateMonitor `to be used as a satelite.
This proposal is not ready to go and would need some cleanup, but shows the idea. If this would be rated as usefull, I plan to integrate the `StateController `portion as well as I did for the `StateIO FW`.

Curious to see your comments.

### CODE REVIEW

**Comment by moinmoin-sh** [comment]
After some thinking I come to the conclusion that this method of integration as addional build target doesn't feel right. Several `#if defines ` make the code hard to follow and maintain.
I think it would be better to create a fresh project that doesn't carry all of the RACK32 overhead like Screen, Ethernet, I2C, Temp sensor. It might be better if I create a new `OXRS-IO-GPIO-ESP32-LIB ` which replaces the `OXRS-SHA-RACK32-ESP32-LIB` and supports only the generic builtin ESP32 HW whilest the API stays compatible to support the OXRS eco system.

**Comment by sumnerboy12** [comment]
Yeah I tend to agree, too many `#if defines` makes the code harder to follow. I do think adding more hardware platform libs is a good idea tho. But in this instance I wonder if we just need a `OXRS-IO-StateMonitorIO-ESP32-FW` (or similar) which can run on any of the supported hardware platforms, but doesn't use I2C or the MCPs?

**Comment by moinmoin-sh** [comment]
My recent plan is (what I'm working on) to have a single new FW (new repro under OXRS-IO) lets say `OXRS-IO-StateIOGpio-ESP32-FW` with it's own main.cpp to run on a base ESP32 without any additional HW. The 16 mentioned GPIOs can be devided into an input and an output partition. (at runtime similar to StateIO)
There will be also a supporting LIB (another new repro under OXRS-IO) like `OXRS-IO-Gpio32-ESP32-LIB` that replaces the `Rack32-ESP32 lib` to support the API to the OXRS eco system.
I think that follows pretty much our concept.
If you have preferences for the FW / LIB namings it would be a good time to let me know now.

I think I'll close this and we can discuss on Discord.

**Comment by moinmoin-sh** [comment]
Not going to be merged in this form.
